### PR TITLE
[sparkbench/ml] rework code structure to support ML/MLLib and Add LR in ML

### PR DESCRIPTION
**Pull request by VinceShieh**

### CODE REVIEW

**Comment by VinceShieh** [comment]
@carsonwang @sophia-sun @heyu1 
please take a look since it's not a minor change :)
thanks.

**Comment by heyu1** [comment]
This PR build failed. I have fixed some issues of travis in #497, please re-submit this PR to trigger the Travis build and test. @VinceShieh

**Comment by heyu1** [comment]
@VinceShieh It build failed in spark 1.6. Please help to re-create a dir for LR workload that using spark.ml which supports DataFrame-based API, so I can add a dedicated pom file for it to avoid its build in spark 1.6.

**Comment by VinceShieh** [comment]
already in there.

**Comment by heyu1** [comment]
I have created a PR "Update pom.xml #1" in your fork "VinceShieh:lr_rework_reconstructured", please help to check if this could help to build pass. Thanks! @VinceShieh

File: sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/LogisticRegression.scala [hunk 1]
```diff
@@ -20,47 +20,38 @@ package com.intel.hibench.sparkbench.ml
...
-    val splits = data.randomSplit(Array(0.6, 0.4), seed = 11L)
-    val training = splits(0).cache()
-    val test = splits(1)
-
+    val inputPath = args(0)
+    val numFeatures = args(1).toInt
+    val aggDepth = if (args.length >2) {
+        if (args(2).toInt < 2) 2 else args(2).toInt
+    } else 0
+ //  val sqlContext= new org.apache.spark.sql.SQLContext(sc)
```
Comment (chenghao-intel) [inline sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/LogisticRegression.scala]: remove the unused code.

File: sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/LogisticRegression.scala [hunk 2]
```diff
@@ -20,47 +20,38 @@ package com.intel.hibench.sparkbench.ml
...
+//   import sqlContext.implicits._
+   val spark = SparkSession.builder().appName("LogisticRegression").getOrCreate()
+   val df = spark.read.parquet(args(0))
+/*
```
Comment (chenghao-intel) [inline sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/LogisticRegression.scala]: same here.

File: sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/LogisticRegression.scala [hunk 3]
```diff
@@ -20,47 +20,38 @@ package com.intel.hibench.sparkbench.ml
...
-
-    val accuracy = predictionAndLabels.filter(x => x._1 == x._2).count().toDouble / predictionAndLabels.count()
-    println(s"Accuracy = $accuracy")
-
-    sc.stop()
+    val model = new LogisticRegression()
+      .setMaxIter(30)
+      .setRegParam(0.3)
+      .setElasticNetParam(0.8)
+// uncomment below config for spark 2.1 or above
```
Comment (chenghao-intel) [inline sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/LogisticRegression.scala]: is there any solution to enable both Spark 2.1 and 2.2? we cannot reply on end user will change the code by following your comments in the code.

File: sparkbench/mllib/src/main/scala/com/intel/sparkbench/mllib/LogisticRegression.scala [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    // $example on$
+    // Load training data in LIBSVM format.
+    val data: RDD[LabeledPoint] = sc.objectFile(inputPath)
+
+    // Split data into training (60%) and test (40%).
+    val splits = data.randomSplit(Array(0.6, 0.4), seed = 11L)
+    val training = splits(0).cache()
+    val test = splits(1)
+println("****vince:" + inputPath +"," + training.count)
```
Comment (chenghao-intel) [inline sparkbench/mllib/src/main/scala/com/intel/sparkbench/mllib/LogisticRegression.scala]: indent

File: sparkbench/mllib/src/main/scala/com/intel/sparkbench/mllib/LogisticRegressionDataGenerator.scala [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+import scala.util.Random
+
+import org.apache.spark.{SparkConf, SparkContext}
+import org.apache.spark.annotation.{DeveloperApi, Since}
+import org.apache.spark.mllib.linalg.Vectors
+import org.apache.spark.mllib.regression.LabeledPoint
+import org.apache.spark.rdd.RDD
+
+/**
+ * :: DeveloperApi ::
```
Comment (chenghao-intel) [inline sparkbench/mllib/src/main/scala/com/intel/sparkbench/mllib/LogisticRegressionDataGenerator.scala]: Remove this annotation?

File: stormbench/streaming/dependency-reduced-pom.xml [hunk 6]
```diff
@@ -0,0 +1,85 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (chenghao-intel) [inline stormbench/streaming/dependency-reduced-pom.xml]: Why do you change the `pom.xml` for stormbench? do they have dependency? can we do it in another PR if you feel some bugs in stormbench?

Comment (VinceShieh) [reply stormbench/streaming/dependency-reduced-pom.xml]: it's a typo.

File: sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/LogisticRegression.scala (hunk 3 above)
Comment (VinceShieh) [reply sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/LogisticRegression.scala]: currently I cannot think of any solution to support this change, suggest removing this change.

**Comment by VinceShieh** [comment]
this PR is obsolete. By far more than 4 new models are merged since this PR.
Suggest re-submit a new PR after 7.0 release.

File: sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/LogisticRegression.scala (hunk 3 above)
Comment (chenghao-intel) [reply sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/LogisticRegression.scala]: `if (spark.SPARK_VERSION.startWith("2.2" ) {
..
} else if (..) {
}
`

**Comment by VinceShieh** [comment]
re-submit the PR after 7.0 release, close for now.
